How to upgrade a baker (#509)
trilitech/tezos-developer-docs: docs/tutorials/join-dal-baker/run-baker.md

      - Inject a consensus attestation for it (log message: "injected attestation ... for my_baker (&lt;address&gt;) for level ..., round ...")
      - Attach a DAL attestation to it, indicating which of the shards assigned to the baker have been seen on the DAL network (log message: "ready to attach DAL attestation for level ..., round ..., with bitset ... for my_baker (&lt;address&gt;) to attest slots published at level ...")

## Upgrading the baker

The version of the baker program depends on the version of the Tezos protocol.
Therefore, when a new version of the Tezos protocol becomes active, you must start the baker for the new protocol immediately.

To simplify the upgrade process, you can follow these steps when the new protocol is about to be activated:

> 1. Check posts on the forum at https://forum.tezosagora.org to see if the new protocol requires a specific version of the Octez suite and upgrade your Octez suite if necessary.

Reviewer: ```suggestion
1. Check the release pages in the [Octez documentation](https://tezos.gitlab.io) (section `Changes in Octez releases`) or check the posts on the forum at https://forum.tezosagora.org to see which version of the Octez suite supports the upcoming protocol and upgrade your Octez suite if necessary.
```

> Announcement posts on the forum may also give instructions for things you need to do for new versions of the Octez suite.

Reviewer: ```suggestion
The Octez release page gives instructions for upgrading.
```

      - Inject a consensus attestation for it (log message: "injected attestation ... for my_baker (&lt;address&gt;) for level ..., round ...")
      - Attach a DAL attestation to it, indicating which of the shards assigned to the baker have been seen on the DAL network (log message: "ready to attach DAL attestation for level ..., round ..., with bitset ... for my_baker (&lt;address&gt;) to attest slots published at level ...")

## Upgrading the baker

The version of the baker program depends on the version of the Tezos protocol.
Therefore, when a new version of the Tezos protocol becomes active, you must start the baker for the new protocol immediately.

To simplify the upgrade process, you can follow these steps when the new protocol is about to be activated:

1. Check the release pages in the [Octez documentation](https://tezos.gitlab.io) (section `Changes in Octez releases`) or check the posts on the forum at https://forum.tezosagora.org to see which version of the Octez suite supports the upcoming protocol and upgrade your Octez suite if necessary.
The Octez release page gives instructions for upgrading.

> 1. Leave the baker for the previous protocol running, such as the `octez-baker-PsParisC` daemon.

Reviewer 2: ```suggestion
2. Leave the baker for the previous protocol running, such as the `octez-baker-PsParisC` daemon.
```

Author: There's no need to manually number  the steps; in markdown we can indicate all numbered steps with `1.` and it will write the numbers. It saves us from renumbering things when we add or remove steps.

Reviewer 2: Oh that's nice!

> 1. Start the baker for the new protocol, such as the `octez-baker-PsQuebec` daemon.

Reviewer 2: ```suggestion
3. Start the baker for the new protocol, such as the `octez-baker-PsQuebec` daemon.
```

This daemon is not yet able to bake because it is using the future version of the protocol, but you can run it early without causing any problems.
However, make sure not to run the baker twice **for the same protocol version and the same baker account**, to avoid being slashed for double signing.

> 1. When the new version of the protocol becomes active, the previous protocol baker is no longer able to bake and the new protocol baker begins to bake automatically.

Reviewer 2: ```suggestion
4. When the new version of the protocol becomes active, the previous protocol baker is no longer able to bake and the new protocol baker begins to bake automatically.
```

> 1. Then you can stop the previous protocol baker.

Reviewer 2: ```suggestion
5. Then you can stop the previous protocol baker.
```
